feat(chat): add support for several sso providers with same type (Issue #4495)

Gimir opened the pull request:
Issues:

- Issue #4495 

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

Commit:
feat(chat): add ability to use several providers with same type

Commit:
Merge branch 'development' of https://github.com/epam/ai-dial-chat into fix/4495-same-type-providers

Gimir commented:
/deploy-review
>GitHub actions run: [19747454725](https://github.com/epam/ai-dial-ci/actions/runs/19747454725)
>Environment URL: [review-environment](https://chat-ai-dial-chat-pr-5187.gke.test.dial.parts) | [pipeline](https://gitlab.deltixhub.com/Clients/dial/test/review/-/pipelines/731724)
>Chat E2E tests status: [failure](https://github.com/epam/ai-dial-ci/actions/runs/19747454725)
>Overlay E2E tests status: [failure](https://github.com/epam/ai-dial-ci/actions/runs/19747454725)

Gimir commented:
/deploy-review
>GitHub actions run: [19758764352](https://github.com/epam/ai-dial-ci/actions/runs/19758764352)
>Environment URL: [review-environment](https://chat-ai-dial-chat-pr-5187.gke.test.dial.parts) | [pipeline](https://gitlab.deltixhub.com/Clients/dial/test/review/-/pipelines/731951)
>Chat E2E tests status: [failure](https://github.com/epam/ai-dial-ci/actions/runs/19758764352)
>Overlay E2E tests status: [success](https://github.com/epam/ai-dial-ci/actions/runs/19758764352)

Commit:
feat(chat): update readme with same type sso's instructions

IlyaBondar commented on apps/chat/src/utils/auth/auth-providers.ts:
```diff
@@ -9,174 +9,282 @@ import OktaProvider from 'next-auth/providers/okta';
+        authorization: {
+          params: {
+            scope:
+              process.env.AUTH_AZURE_AD_SCOPE ||
```
can be different scope for `AUTH_AZURE_1_AD_SCOPE`

IlyaBondar commented on apps/chat/src/utils/auth/auth-providers.ts:
```diff
@@ -9,174 +9,282 @@ import OktaProvider from 'next-auth/providers/okta';
+        authorization: {
+          params: {
+            scope:
+              process.env.AUTH_AZURE_AD_SCOPE ||
```
can be different scope for `AUTH_AZURE_AD_1_SCOPE`

Commit:
feat(chat): add redirect uri example

IlyaBondar commented on apps/chat/src/utils/auth/auth-providers.ts:
```diff
@@ -9,174 +9,282 @@ import OktaProvider from 'next-auth/providers/okta';
+        authorization: {
+          params: {
+            scope:
+              process.env.AUTH_GOOGLE_SCOPE ||
```
the same

IlyaBondar commented on apps/chat/src/utils/auth/auth-providers.ts:
```diff
@@ -9,174 +9,282 @@ import OktaProvider from 'next-auth/providers/okta';
+};
+
+const getProviderConfig = (provider: SupportedProviders, index = 0) => {
+  const config = {
```
can be simplified to read
```suggestion
  const getEnv = (name:string) =>  getProviderEnv(provider, name, index);
  const config = {
```
and then
```
getEnv(ProviderConfigFields.CLIENT_ID),
getEnv(ProviderConfigFields.CLIENT_SECRET)
```

Commit:
fix(chat): fix passing indexed scope env

Commit:
fix(chat): refactor

IlyaBondar commented on apps/chat/src/utils/auth/auth-providers.ts:
```diff
@@ -9,174 +9,278 @@ import OktaProvider from 'next-auth/providers/okta';
+    default:
+      return null;
+  }
+};
```
```suggestion
const providerConfigMethods = {
  [SupportedProviders.AUTH0]: getAuth0Provider,
  [SupportedProviders.AZURE_B2C]: getAzureB2CProvider,
  [SupportedProviders.AZURE_AD]: getAzureProvider,
  [SupportedProviders.COGNITO]: getCognitoProvider,
  [SupportedProviders.GOOGLE]: getGoogleProvider,
  [SupportedProviders.KEYCLOAK]: getKeycloakProvider,
  [SupportedProviders.OKTA]: getOktaProvider,
  [SupportedProviders.GITLAB]: getGitLabProvider,
  [SupportedProviders.PING_ID]: getPingIdProvider,
};
const getProviderFromConfig = (config: ProviderConfig) => providerConfigMethods[config.provider]?.(config);
```

Commit:
fix(chat): refactor

sr-remsha committed:
Revise README for clarity on auth provider configurations

Updated the README.md to improve clarity and consistency in the documentation regarding multiple configurations for authentication providers.

Commit:
fix(chat): refactor

Commit:
Merge branch 'fix/4495-same-type-providers' of https://github.com/epam/ai-dial-chat into fix/4495-same-type-providers

Commit:
Merge branch 'development' of https://github.com/epam/ai-dial-chat into fix/4495-same-type-providers

Commit:
fix(chat): run formatter

Gimir commented:
/deploy-review
>GitHub actions run: [19764425007](https://github.com/epam/ai-dial-ci/actions/runs/19764425007)
>Environment URL: [review-environment](https://chat-ai-dial-chat-pr-5187.gke.test.dial.parts) | [pipeline](https://gitlab.deltixhub.com/Clients/dial/test/review/-/pipelines/732243)
>Chat E2E tests status: [success](https://github.com/epam/ai-dial-ci/actions/runs/19764425007)
>Overlay E2E tests status: [success](https://github.com/epam/ai-dial-ci/actions/runs/19764425007)
